### PR TITLE
gh-101282: Revert some M4 style for BOLT configuration

### PR DESCRIPTION
**Pull request by corona10**
* Issue: gh-101282

Without this revert on macOS:
````
checking for --enable-bolt... no
checking BOLT_INSTRUMENT_FLAGS...
checking BOLT_APPLY_FLAGS... ./configure: line 8401: -reorder-blocks=ext-tsp: command not found

````

With this revert on macOS:
````
checking BOLT_INSTRUMENT_FLAGS...
checking BOLT_APPLY_FLAGS... -update-debug-sections -reorder-blocks=ext-tsp -reorder-functions=hfsort+ -split-functions -icf=1 -inline-all -split-eh -reorder-functions-use-hot-size -peepholes=none -jump-tables=aggressive -inline-ap -indirect-call-promotion=all -dyno-stats -use-gnu-stack -frame-opt=hot

````

### CODE REVIEW

**Review by ned-deily — APPROVED** [review]
LGTM, thanks!

**Comment by corona10** [comment]
@ned-deily cc @erlend-aasland 

I am preparing another PR too.

**Comment by corona10** [comment]
cc @indygreg  Not decided yet which PR will be merged

File: configure.ac [hunk 1]
```diff
@@ -2085,26 +2085,21 @@ AC_ARG_VAR(
...
+ -inline-all\
+ -split-eh\
+ -reorder-functions-use-hot-size\
+ -peepholes=none\
+ -jump-tables=aggressive\
+ -inline-ap\
+ -indirect-call-promotion=all\
+ -dyno-stats\
+ -use-gnu-stack\
+ -frame-opt=hot"
```
Comment (erlend-aasland) [inline configure.ac]:
```suggestion
  BOLT_APPLY_FLAGS="\
 -update-debug-sections\
 -reorder-blocks=ext-tsp\
 -reorder-functions=hfsort+\
 -split-functions\
 -icf=1\
 -inline-all\
 -split-eh\
 -reorder-functions-use-hot-size\
 -peepholes=none\
 -jump-tables=aggressive\
 -inline-ap\
 -indirect-call-promotion=all\
 -dyno-stats\
 -use-gnu-stack\
 -frame-opt=hot\
"
```

**Review by erlend-aasland — CHANGES_REQUESTED** [review]
If we go for this variant, I'd make the list more "git diff" friendly. I left a suggestion for this.

**Review by erlend-aasland — APPROVED** [review]
LGTM, but I prefer #104752

**Pull request closed without merging**